### PR TITLE
ORC-514: [C++] Malformed RowIndex stream meta exception when read statistics from stripe with bloom filter

### PR DESCRIPTION
**Pull request by rip-nsk**

### CODE REVIEW

File: c++/src/Reader.cc [hunk 1]
```diff
@@ -530,6 +531,9 @@ namespace orc {
           (*indexStats)[column].push_back(entry.statistics());
         }
       }
+      if (streamKind != StreamKind::StreamKind_ROW_INDEX) {
```
Comment (omalley) [inline c++/src/Reader.cc]: Can you make this an else?

Comment (rip-nsk) [reply c++/src/Reader.cc]: sure.
is WriterImpl::writeStripe() works correctly? probably bloom filter streams should be added to the indexLength, not dataLength.

Comment (wgtmac) [reply c++/src/Reader.cc]: You are right. WriterImpl::writeStripe() should be fixed to include bloom filter index.

File: c++/src/Reader.cc [hunk 2]
```diff
@@ -529,6 +530,11 @@ namespace orc {
           const proto::RowIndexEntry& entry = rowIndex.entry(j);
           (*indexStats)[column].push_back(entry.statistics());
         }
+      } else if (streamKind == StreamKind::StreamKind_BLOOM_FILTER_UTF8) {
+        // ORC-514
```
Comment (wgtmac) [inline c++/src/Reader.cc]: I'd suppose to fix those affected files instead of doing these here.

**Pull request merged**